### PR TITLE
fix: return friendly message when kafka producer and consumer fails to start (rv5.0)

### PR DESCRIPTION
**Pull request by thalesmg**
## Note: targeting `release-50`

Fixes https://emqx.atlassian.net/browse/EMQX-9392

The returned information does not allow to diagnose the issue (i.e.: a connection issue due to the wrong host and port, the wrong password failing authn).  However, such information is printed to the logs.

This changes the returned error to the API so that the user is hinted at looking at the logs for further investigation of the error.

#### Before the fix

image

#### After the fix

image

When probing:

image

image

image

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` and `.zh.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Comment by zmstone** [comment]
for throws, we can log the stacktrace but no need to expose to user ?

**Comment by thalesmg** [comment]
> for throws, we can log the stacktrace but no need to expose to user ?

Good point, I'll adapt the point where the throws are collected for such cases.

**Comment by thalesmg** [comment]
I've made a similar change for the Kafka consumer error message here.

File: lib-ee/emqx_ee_bridge/src/kafka/emqx_bridge_impl_kafka_consumer.erl [hunk 1]
```diff
@@ -95,6 +95,11 @@
     commit_fun => brod_group_subscriber_v2:commit_fun()
 }.
 
+-define(CLIENT_DOWN_MESSAGE,
+    "Failed to start Kafka client. Please check the logs for errors and check"
+    " the connection parameters"
```
Comment (zmstone) [inline lib-ee/emqx_ee_bridge/src/kafka/emqx_bridge_impl_kafka_consumer.erl:100]:
```suggestion
-define(CLIENT_DOWN_MESSAGE,
    "Failed to start Kafka client. Please check the logs for errors and check"
    " the connection parameters."
```

File: apps/emqx_resource/src/emqx_resource.erl [hunk 2]
```diff
@@ -356,7 +356,14 @@ is_buffer_supported(Module) ->
...
     {ok, resource_state()} | {error, Reason :: term()}.
 call_start(MgrId, Mod, Config) ->
-    ?SAFE_CALL(Mod:on_start(MgrId, Config)).
+    try
+        Mod:on_start(MgrId, Config)
+    catch
+        throw:{error, Error} ->
+            {error, Error};
+        Kind:Error:Stacktrace ->
+            {error, {Kind, Error, Stacktrace}}
```
Comment (zmstone) [inline apps/emqx_resource/src/emqx_resource.erl:365]:
```suggestion
            {error, #{exception => Kind, reason => Error, stacktrace => Stacktrace}}
```

File: apps/emqx_resource/src/emqx_resource.erl [hunk 3]
```diff
@@ -356,7 +356,14 @@ is_buffer_supported(Module) ->
 -spec call_start(manager_id(), module(), resource_config()) ->
     {ok, resource_state()} | {error, Reason :: term()}.
 call_start(MgrId, Mod, Config) ->
-    ?SAFE_CALL(Mod:on_start(MgrId, Config)).
+    try
+        Mod:on_start(MgrId, Config)
+    catch
+        throw:{error, Error} ->
+            {error, Error};
```
Comment (zmstone) [inline apps/emqx_resource/src/emqx_resource.erl:363]:
```suggestion
        throw:Reason ->
            {error, Reason};
```

**Review by zmstone — APPROVED** [review]

File: lib-ee/emqx_ee_bridge/src/kafka/emqx_bridge_impl_kafka_consumer.erl [hunk 4]
```diff
@@ -173,7 +178,7 @@ on_get_status(_InstanceID, State) ->
         kafka_client_id := ClientID,
         kafka_topics := KafkaTopics
     } = State,
-    do_get_status(ClientID, KafkaTopics, SubscriberId).
+    do_get_status(State, ClientID, KafkaTopics, SubscriberId).
```
Comment (savonarola) [inline lib-ee/emqx_ee_bridge/src/kafka/emqx_bridge_impl_kafka_consumer.erl:181]: Nit: we pass `State` just to get it back unchanged.. 
Mb we could decorate the result with state after the call 🤔

Comment (thalesmg) [reply lib-ee/emqx_ee_bridge/src/kafka/emqx_bridge_impl_kafka_consumer.erl:181]: I'll refactor those points in a follow up PR.

**Pull request merged**